src/Optimizer.cc

```diff
@@ -1101,6 +1101,9 @@ int Optimizer::PoseOptimization(Frame *pFrame)
 
         if(optimizer.edges().size()<10)
             break;
+        
+        if((nInitialCorrespondences-nBad)<5)
```

Comment:
sort of a separate issue, but this prevents the `virtual int g2o::SparseOptimizer::optimize(int, bool): 0 vertices to optimize, maybe forgot to call initializeOptimization()` errors from appearing. Having that log get spammed usually meant the segfault would occur when saving a map but adding this fix did not prevent the errors. Leaving this in appears to make the` TRACK_REF_KF: Less than 15 matches!!` appear more. not sure which is better for performance and can remove this if we want

Reply:
got the fix from [here](https://github.com/raulmur/ORB_SLAM2/issues/211#issuecomment-267534863)

Reply:
I think we should remove this if we're not sure how it changes the behavior.